src/libraries/System.Private.CoreLib/src/System/IO/FileSystemInfo.cs

```diff
@@ -107,6 +115,41 @@ public DateTime LastWriteTimeUtc
             set => LastWriteTimeCore = File.GetUtcDateTimeOffset(value);
         }
 
+        /// <summary>
+        /// If this <see cref="FileSystemInfo"/> instance represents a link, returns the link target's path.
+        /// If a link does not exist in <see cref="FullName"/>, or this instance does not represent a link, returns <see langword="null"/>.
+        /// </summary>
+        public string? LinkTarget => _linkTarget ??= FileSystem.GetLinkTarget(FullPath, this is DirectoryInfo);
+
+        /// <summary>
+        /// Creates a symbolic link located in <see cref="FullName"/> that points to the specified <paramref name="pathToTarget"/>.
+        /// </summary>
+        /// <param name="pathToTarget">The path of the symbolic link target.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="pathToTarget"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="pathToTarget"/> is empty.
+        /// -or-
+        /// This instance was not created passing an absolute path.
+        /// -or-
+        /// <paramref name="pathToTarget"/> contains invalid path characters.</exception>
+        /// <exception cref="IOException">A file or directory already exists in the location of <see cref="FullName"/>.
+        /// -or-
+        /// An I/O error occurred.</exception>
+        public void CreateAsSymbolicLink(string pathToTarget)
+        {
+            FileSystem.VerifyValidPath(pathToTarget, nameof(pathToTarget));
+            FileSystem.CreateSymbolicLink(OriginalPath, pathToTarget, this is DirectoryInfo);
+            Invalidate();
```

Comment:
Do we really need to invalidate?

Reply:
Yes. It's consistent to the `Create` methods:
- [FileInfo.cs#L105](https://github.com/dotnet/runtime/blob/44f050acc0f7791d6cd7ac772945444912bcf299/src/libraries/System.Private.CoreLib/src/System/IO/FileInfo.cs#L105)
- [DirectoryInfo.cs#L95](https://github.com/dotnet/runtime/blob/44f050acc0f7791d6cd7ac772945444912bcf299/src/libraries/System.Private.CoreLib/src/System/IO/DirectoryInfo.cs#L95)

Since this method is part of an _instance_ that represents a filesystem entry, there could be some time between the creation of the link and actual consultation of an attribute. During that time, the symlink could've changed (renamed, deleted, etc.), so the next time the user calls an attribute, a full refresh will be required to ensure the latest info is retrieved.